Display the Galaxy Citation within the Export Tool References List

mschatz opened the pull request:
This commit implements a new feature to display a user-defined Galaxy Citation within the "Export Tool References List". By default, this includes a reference to the NAR paper, but this can be changed at runtime via a galaxy.yml edit or by rebuilding the client with an updated config_schma.yml. This feature was proposed and discussed here: https://github.com/galaxyproject/galaxy/issues/19362

Summary of changes:
- Define a new configuration property for citation_bibtex with a reference to the current main galaxy paper.
- Inject that as a citation within the Export Tool Refernences to diplay alongside other tools used in the current history.
- Uses the Cite() package to format in bibtex or APA as needed (as done for all other tools too)
- Updates the docs, and schema as needed to provide the default value and description

Example showing Galaxy citation along with several others
<img width="1789" alt="image" src="https://github.com/user-attachments/assets/db595d6f-1996-4745-a7ca-db8e45003fb7" />

Example with a blank history just showing Galaxy
<img width="1789" alt="image" src="https://github.com/user-attachments/assets/97f93f49-9eff-41b5-94c2-45b94bf02f34" />


## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [X] Instructions for manual testing are as follows:
  1. Load/create a history
  2. Select "Export Tool References" from the History options hamburger menu

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

mschatz committed:
Display the Galaxy Citation within the Export Tool References List

This commit implements a new feature to display a user-defined Galaxy Citation
within the "Export Tool References List". By default, this includes
a reference to the NAR paper, but this can be changed at runtime
via a galaxy.yml edit or by rebuilding the client with an updated
config_schma.yml. This feature was proposed and discussed here:
https://github.com/galaxyproject/galaxy/issues/19362

Summary of changes:
- Define a new configuration property for citation_bibtex with a reference to the current main galaxy paper.
- Inject that as a citation within the Export Tool Refernences to diplay alongside other tools used in the current history.
- Uses the Cite() package to format in bibtex or APA as needed (as done for all other tools too)
- Updates the docs, and schema as needed to provide the default value and description

ahmedhamidawan commented on client/src/components/Citation/services.ts:
```diff
@@ -1,8 +1,9 @@
-import { rethrowSimple } from "@/utils/simple-error";
 import { useConfig } from "@/composables/config";
+import { rethrowSimple } from "@/utils/simple-error";
+
```
```suggestion
```

The extra line added here is triggering the lint error again 😁

You can run `yarn eslint . --fix` to autofix these. If VSCode is your choice of IDE, the [Eslint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) catches these inline as well!

mschatz replied:
Thank you. I just figured that out here too!

ahmedhamidawan replied:
Similarly, we have the `make format` command which autofixes several backend linting errors.

ahmedhamidawan replied:
And the `tox -e mypy` command for mypy errors!

mschatz committed:
clean up linting errors and missing api functions

ahmedhamidawan committed:
move Galaxy config citation to `CitationsList` directly instead of services

We decided that it makes more sense for the Galaxy citation to be shown in the `Export Tool References` view, rather than on each tool form as well.

ahmedhamidawan reviewed (approved):
Very nice addition, thank you very much @mschatz!

This initially showed the Galaxy citation on each tool form as well, but we decided that it makes more sense to only show it in the tool references list (as shown in the main PR description screenshots). In the future, maybe we can decide to show the Galaxy citation in a separate section on tool forms as well 🤷 ?

Given that this is a config value, we could also change this to a `string[]` instead of a singular `string`, to allow for multiple citations configurable per server, in the future.
